feat: Added a `--merge-state` flag to `meltano run` to merge state with that of previous runs (#8195)
meltano/meltano: docs/docs/reference/command-line-interface.md

- One or more `--select <entity>` options can be passed to only extract records for matching [selected entities](#select).
  Similarly, `--exclude <entity>` can be used to extract records for all selected entities _except_ for those specified.

> - A `--merge-state` flag can be passed to merge or overwrite the existing state at the end of the run.

Reviewer: ```suggestion
- A `--merge-state` flag can be passed to merge state with that of previous runs.
```

- `--full-refresh` will force a full refresh and ignore the prior state. The new state after completion will still be updated with the execution results, unless `--no-state-update` is also specified.
- `--force` will force a job run even if a conflicting job with the same generated ID is in progress.
- `--state-id-suffix` define a custom suffix to generate a state ID with for each EL pair.

> - `--merge-state` will merge or overwrite the existing state at the end of the run .

Reviewer: ```suggestion
- `--merge-state` will merge state with that of previous runs.
```

# run a pipeline with a custom state ID suffix
# the autogenerated ID for the EL pair will be 'dev:tap-gitlab-to-target-postgres:pipeline-alias'
meltano --environment=dev --state-id-suffix pipeline-alias run tap-gitlab hide-secrets target-postgres

> # run the pipelines in series, performing a merging of state at the end of run.
> meltano --environment=dev run --merge-state tap-gitlab target-postgres tap-salesforce target-mysql ...

Reviewer: This two features mentioned in tandem may lead to confusion, e.g. "is it gonna merge state from the two pipelines together?".

A simpler example would be more clear:

```suggestion
# run a pipeline, merging state with that of previous runs.
meltano --environment=dev run --merge-state tap-gitlab target-postgres
```
